lib/lvol/lvol.c

```diff
@@ -1225,14 +1295,19 @@ create_lvol_snapshot(struct spdk_lvol *origlvol, const char *snapshot_name,
 		cb_fn(cb_arg, NULL, rc);
 		return;
 	}
-
+	snap_req = calloc(1, sizeof(*snap_req));
+	if (!snap_req) {
+		SPDK_ERRLOG("Cannot alloc memory for lvol snapshot request pointer\n");
+		cb_fn(cb_arg, NULL, -ENOMEM);
+		return;
+	}
 	req = calloc(1, sizeof(*req));
 	if (!req) {
 		SPDK_ERRLOG("Cannot alloc memory for lvol request pointer\n");
 		cb_fn(cb_arg, NULL, -ENOMEM);
```

Comment:
We also need to deallocate 'snap_req' here.

Reply:
Fixed. One more deallocation in failure of lvol acllocation is also added.